src/io/flutter/sdk/FlutterSdkVersion.java

```diff
@@ -48,62 +27,55 @@ private FlutterSdkVersion(@NotNull String versionString) {
 
   @NotNull
   public static FlutterSdkVersion readFromSdk(@NotNull VirtualFile sdkHome) {
-    final File versionFile = new File(sdkHome.getPath() + "/VERSION");
-
-    // Use the cache if the file's last modfication date didn't change.
-    // But we still stat the file every time this is called. Not sure the cache saves us much?
-    // TODO(skybrian) should we use the VirtualFile cache instead?
-    final Pair<File, Long> key = Pair.create(versionFile, versionFile.lastModified());
-    if (cache.containsKey(key)) {
-      return cache.get(key);
+    final VirtualFile file = sdkHome.findChild("version");
+
+    if (file == null) {
+      return MIN_SUPPORTED_SDK;
     }
 
-    final String versionString = readVersionFromFile(versionFile);
+    final String versionString = readVersionString(file);
+
     if (versionString == null) {
-      LOG.warn("Unable to find Flutter SDK version at " + sdkHome.getPath());
+      return MIN_SUPPORTED_SDK;
     }
 
-    // If we don't have a version file at all, assume it's a supported version and don't complain.
-    final FlutterSdkVersion version = versionString == null ? MIN_SUPPORTED_SDK : new FlutterSdkVersion(versionString);
+    return new FlutterSdkVersion(versionString);
+  }
+
+  private static String readVersionString(VirtualFile file) {
+    try {
+      final String data = new String(file.contentsToByteArray(), StandardCharsets.UTF_8);
+      for (String line : data.split("\n")) {
+        line = line.trim();
+
+        if (line.isEmpty() || line.startsWith("#")) {
+          continue;
+        }
 
-    cache.put(Pair.create(versionFile, versionFile.lastModified()), version);
-    return version;
+        return line;
+      }
+      return null;
+    }
+    catch (IOException e) {
+      return null;
+    }
   }
 
-  public boolean isSupported() {
+  public boolean isMinRecommendedSupported() {
     return version.compareTo(MIN_SUPPORTED_SDK.version) >= 0;
```

Comment:
If these three are implemented in the same way, the other two methods, flutterTestSupportsMachineMode() and  flutterTestSupportsFiltering() should just return the result of this one, instead of C&P.

Reply:
👍